modal/image.py

```diff
@@ -650,12 +661,44 @@ def add_local_dir(self, local_path: Union[str, Path], remote_path: str, *, copy:
         copy=True can slow down iteration since it requires a rebuild of the Image and any subsequent
         build steps whenever the included files change, but it is required if you want to run additional
         build steps after this one.
+
+        **Usage:**
+
+        ```python
+        from modal._utils.pattern_matcher import PatternMatcher
```

Comment:
Lets put PatternMatcher (or whatever we want to call it) in a publicly importable module (no underscores), exported through `modal.__init__`. 

Otherwise we'd start breaking user's code when we refactor these internals

Reply:
See comment above about naming this class too (if we're moving it to a user importable, we need to make the name more specific)

Reply:
Moved it!